client/components/site-selector/use-onboarding-url.ts

```diff
@@ -0,0 +1,35 @@
+import config from '@automattic/calypso-config';
+import { useLocale } from '@automattic/i18n-utils';
+import { useExperiment } from 'calypso/lib/explat';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+
+const GUTENBOARDING_LOCALES = [ 'en', 'en-gb' ];
+
+/**
+ * Returns the onboarding URL.
+ *
+ * @returns {string}  URL of the onboarding flow for existing users.
+ */
+const useOnboardingUrl = (): string => {
```

Comment:
Seems a bit strange to me that we are assigning an experiment when we are generating a URL.

I think what might be useful here is a redirect experiment - if the user navigates to `config( 'gutenboarding_url' );` we check experiment eligibility and redirect to `config( 'signup_url' )` if treatment. 

Ideally assignment would occur on-click or on the next page we would redirect into the experiment if it does need to have a url.

Reply:
In this way, when the user navigates to `config('gutenboarding_url')`, they may see the loading screen and wait for the experiment result. After the experiment result is ready, they'll redirect to `config( 'signup_url' )` and might need to wait for loading again.

Besides, if we redirect to `config( 'gutenboarding_url' )` first and redirect to `config( 'signup_url' )` later, calypso will load unnessasry resources for `config( 'gutenboarding_url' )` and it's not good for performance. Also, we need to take care about the page view of `config( 'gutenboarding_url' )` or something else to avoid any redundant `config( 'gutenboarding_url' )` page.

Therefore, I prefer the current way as I think it makes it simple and provides a better user experience.

Reply:
Good points, @arthur791004 🙌🏼 

> a better user experience.

Not only that (because we wait for the experiment assignment to load) but also the accuracy of the experiment can be impacted by such redirect. From the previous attempts at running `/new` redirect experiments in signup controller, we've identified 2 main issues: 
* Even worse user experience for `treatment` (it's a completely different entrypoint that needs to be loaded) than for `control` depending on the connection speed and/or device. 
* Bounce rate may impact more the conversion rate for `control`. Even if the bot traffic is filtered, there are still real users that don't wait for the page to load. In this case, the exposure event (`calypso_signup_step_start`) may not even fire for `treatment` but it will fire for `control` that is immediately rendered.

These two may cancel each other sometimes but it's impossible to quantify how much.

However, if we really need to do a redirect experiment, I think there should be a way to preload the assets for both entrypoints before rendering the variation assigned for the user. It will be an even poorer user experience but at least we reduce the margin of error as much as possible.

Reply:
hmm I do see @jessie-ross's point though, since basically every user who loads calypso will call useOnboardingUrl , they will be assigned to an experiment group. But how many of them are going to actually use this link in the UI and how many of them are going to load the `/new` or `/start` flow from some other link? But then we don't want to slow down the user experience by adding a redirect, and especially, we don't want one variation to be significantly slowed down by a redirect while the other isn't.

I think that in this case, we are very motivated to deactivate the /new flow for maintainance and tech strategy reasons, so perhaps running an extremely clean experiment isn't as important as it would be if this were an experimental flow that we were testing. Knowing that, my preference is to go with the current approach, and acknowledge that some users may be incorrectly assigned which will cause noise in the experiment, with the tradeoff that performance is maintained.